src/docfx/config/ops/OpsConfigLoader.cs

```diff
@@ -105,7 +105,9 @@ internal static class OpsConfigLoader
 
             var joinTOCPluginConfig = docsetConfig?.JoinTOCPlugin ?? opsConfig.JoinTOCPlugin ?? Array.Empty<OpsJoinTocConfig>();
             (result["fileMetadata"], result["joinTOC"]) =
-                GenerateJoinTocMetadataAndConfig(joinTOCPluginConfig, new PathString(buildSourceFolder));
+                GenerateJoinTocMetadataAndConfig(
```

Comment:
suggest to prepare `joinTOCPluginConfig` in advance, and simplify the logic of `GenerateJoinTocMetadataAndConfig`

Reply:
fixed.